cw-relayer/relayer/relayer.go

```diff
@@ -116,7 +123,38 @@ func (r *Relayer) setActiveDenomPrices(ctx context.Context) error {
 		return err
 	}
 
+	if queryResponse.ExchangeRates.Empty() {
+		return fmt.Errorf("exchange rates empty")
+	}
+
 	r.exchangeRates = queryResponse.ExchangeRates
+
+	if postMedian {
+		ctx, cancel := context.WithTimeout(ctx, 15*time.Second)
+		defer cancel()
+
+		medianQueryResponse, err := queryClient.Medians(ctx, &oracletypes.QueryMedians{})
+		if err != nil {
+			return err
+		}
+
+		if medianQueryResponse.Medians.Empty() {
+			return fmt.Errorf("median rates empty")
+		}
+
+		deviationsQueryResponse, err := queryClient.MedianDeviations(ctx, &oracletypes.QueryMedianDeviations{})
```

Comment:
@rbajollari @zarazan are these the queries we want to be transferred to the contracts? irrc there was some debate over the format of the available median queries

Reply:
I think this is fine for now based on how the queries currently work but both the median and median deviaton queries will return multiple values per denom. And without block numbers on them we have to rely on the ordering. I think we'll eventually change this and we'll have to update the relayer when we do. Lots of dependencies across these projects 🤯